docs/hooks.md

```diff
@@ -59,7 +59,22 @@ This is used in SystemJS core to ensure that import maps are loaded so that the
 
 This function downloads and executes the code for a module. The promise must resolve with a "register" array, as described in the `getRegister` documentation.
 
-The default system.js implementation is to append a script tag that downloads and executes the module's code, subsequently resolving the promise with the most recent register: `resolve(System.getRegister())`. [Example](https://github.com/systemjs/systemjs/blob/master/src/features/script-load.js).
+The default system.js implementation is to append a script tag that downloads and executes the module's code, subsequently resolving the promise with the most recent register: `resolve(System.getRegister())`. 
+
+It is impossible to have an "identifier" to mark the module identity like AMD therefore SystemJS is using a race condition prone way to register the module. Therefore, you must make sure to maintain the execution order of the code.
```

Comment:
IDs can be enabled via the named register extension, but we discourage this as there is no analog in the modules specification in JavaScript. The mechanism of execution order matches AMD exactly for anonymous definitions.

Reply:
Yeah, the named module is not compatible with ES Module so I didn't mention that.

Reply:
To clarify here, this sentence is wrong "it is impossible" needs to change.

Reply:
Resolved